### DISCUSSION TITLE
Model in Model

### DISCUSSION THREAD

**Discussion by ehtick** [microsoft/maker.js#505] [General]
Is it possible to put a model inside a model, so that the inside model can use constraints from the parent model?

**Comment by danmarshall** [comment]
Yes. The model tree structure does not need to relate to the size of the drawing. Here's an example of a star, which contains a polygon child model, but the polygon is actually larger than its parent (the star):

```js
var makerjs = require('makerjs');

var number_of_sides = 5;
var radius = 70;
var star_outer_radius = 50;
var star_inner_radius = 20;

var poly = new makerjs.models.Polygon(number_of_sides, radius);
var star =  new makerjs.models.Star(number_of_sides, star_outer_radius, star_inner_radius);
  
star.models = {
  poly: poly
};

module.exports = star;
```
